polar-core/src/resource_block.rs

```diff
@@ -167,15 +170,31 @@ pub fn resource_block_from_productions(
     }
 }
 
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum Declaration {
     Role,
     Permission,
     /// `Term` is a `Symbol` that is the (registered) type of the relation. E.g., `Org` in `parent: Org`.
     Relation(Term),
 }
 
-#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+impl fmt::Display for Declaration {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Role => {
+                write!(f, "role")
+            }
+            Self::Permission => {
+                write!(f, "permission")
+            }
+            Self::Relation(_) => {
+                write!(f, "relation")
+            }
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, Hash, PartialEq, Eq)]
```

Comment:
Why do we need all these new `derive`s? Maybe leftover from when you had that other new error variant? In general adding `derive`s increases compile time, so not much reason to add them until we need them.

Reply:
`Serialize, Deserialize` are ✂️ they were necessary only for the duplicate shorthand rule error variant (now gone)